### PR TITLE
Update CI to GitHub Actions

### PR DESCRIPTION
**Pull request by reedy**
cf https://github.com/pear/Net_SMTP/blob/52d04ca9c3b868d201420e8962bd36a30d606780/.github/workflows/ci.yml

### CODE REVIEW

**Comment by reedy** [comment]
https://pear.php.net/bugs/bug.php?id=28662 raises the question of what PHP version etc...

**Comment by reedy** [comment]
CI testing on https://github.com/reedy/Net_URL2/pull/1

No apparent failures... Which seems a little surprisingly maybe :)

**Comment by reedy** [comment]
For https://github.com/pear/Net_URL2/pull/14 we would need to go to PHP 7.1...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up PHP ${{ matrix.php-version }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          ini-values: include_path=.:/usr/share/php:pear/usr/share/php
+      - name: Install dependencies
+        run: |
+          pear -v install -R pear -o Net_Socket
```
Comment (reedy) [inline .github/workflows/ci.yml:28]: Guess this isn't needed

Comment (hakre) [reply .github/workflows/ci.yml]: Just curious, where was this from?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-version: ['5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3', '8.4']
```
Comment (hakre) [inline .github/workflows/ci.yml:17]: IIRC shivammathur/setup-php supports 5.3+ directly so this could ease the maintenance.

Comment (reedy) [reply .github/workflows/ci.yml:17]: https://github.com/shivammathur/setup-php?tab=readme-ov-file#tada-php-support seems to agree, indeed

**Review by ashnazg — APPROVED** [review]

**Pull request merged**